test_tools/joiner/text_converter.cc

```diff
@@ -0,0 +1,147 @@
+#include <string>
+#include <vector>
+#include <fstream>
+#include <flatbuffers/flatbuffers.h>
+#include <iostream>
+#include "../../rlclientlib/logger/preamble.h"
+#include "../../rlclientlib/logger/message_type.h"
+#include "../../rlclientlib/generated/RankingEvent_generated.h"
+#include "../../rlclientlib/generated/OutcomeEvent_generated.h"
+
+// namespace aliases
+namespace rlog = reinforcement_learning::logger;
+namespace flat = reinforcement_learning::messages::flatbuff;
+////
+
+namespace reinforcement_learning { namespace joiner {
+
+  // forward declarations 
+  void convert_to_text(const std::string& file);
+  void print_ranking_event(void* buff);
```

Comment:
consider passing this an `ostream` so that it can be reused for printing to stdout and a file

Reply:
Done.